### ISSUE TITLE
optimize: Combine Global and Local

### ISSUE THREAD

**Issue by btracey** [gonum/gonum#482]
Following some offline discussions with @vladimir-ch , I think `optimize.Global` and `optimize.Local` should be combined into a single call, `optimize.Minimize`. The two function calls use a largely overlapping code and interface at the moment. The two major differences are that 1) Local is only serial 2) Local takes in an initial location while global does not. The second is easily adjusted  by adding a field to Settings, and saying that all 0s is the initial location by default. The first will add a bit of overhead to calls to local, but I think this cost will be made up for by making the optimize package easier to use (a single function rather than two, a single `Method` interface rather than two, etc.), making there be less code overall, and I suspect there will be significant improvement in the legibility of Local methods that comes along.

**Comment by btracey** [comment]
A couple weeks ago I spent a bit of time with this. I decided that doing this all at once would be a massive change that would be extremely difficult to review. I suggest a couple of steps

- Replace the code in Local with a call to Global, and adding a shell `GlobalMethod`. Add an InitLocation field to `GlobalSettings` to allow this to happen
- Remove all of the Local specific code, much of this will just be moving to Global
- Fix the current Local methods to "natively" use the Global interface 
- Delete `Method`, `Settings`, etc. and rename `GlobalMethod` to `Method` etc.
